gufe/ligandnetwork.py

```diff
@@ -184,6 +184,36 @@ def enlarge_graph(self, *, edges=None, nodes=None) -> LigandNetwork:
 
         return LigandNetwork(self.edges | set(edges), self.nodes | set(nodes))
 
+    def remove_edges(self, edges: Union[LigandAtomMapping, list[LigandAtomMapping]]) -> LigandNetwork:
+        """Create a new copy of this network with some edges removed
+
+        Note that this will not remove any nodes, potentially resulting in
```

Comment:
How big of deal is this? Is it worth adding the complexity of checking to make sure we don't allow that? I can't remember what the rules are, if we allow creating a network that is disconnected, then I don't think we need a check, but if we require a connected network, then we want to make sure we don't let users put things into a inconsistent state.

Reply:
This is a good suggestion, I think :)

Reply:
Yeah I was 50/50 on what is best here. Currently disconnected graphs are "allowed" in a data structure sense, but they're often undesirable. It wouldn't be impossible to find and remove orphan nodes, but then the method isn't strictly "remove edges". I think I'll chicken out and add a kwarg that allows either, default being remove orphans.

Reply:
> It wouldn't be impossible to find and remove orphan nodes

Ah yes, I was suggesting a more gentle "throw an error if removing this edge creates an orphan" or something so it can stay focused and do what it says on the tin, so I would be happy with kwargs that control things like, raising an error or not if it would create an orphan, and/or remove orphan(s) (this will be tricking if removing a edge creates 2 disconnect graphs, which one gets remove? the one with more nodes? -- maybe we can support the case where if it creates a single orphan, the default is to raise an error, but that can be suppressed, and additionally with another kwarg the orphan can be removed)